frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -763,8 +768,13 @@ export const experimentLogic = kea<experimentLogicType>([
         },
         updateExperimentSuccess: async ({ experiment }) => {
             actions.updateExperiments(experiment)
-            actions.loadMetricResults()
-            actions.loadSecondaryMetricResults()
+            if (experiment.start_date) {
+                actions.loadMetricResults()
+                actions.loadSecondaryMetricResults()
+            } else {
+                actions.loadMetricResultsSuccess([])
```

Comment:
hmm, not sure I understand. Maybe I wasn't clear.

Took a look and it doesn't look like there's any other listeners listening on `loadMetricResultsSuccess`. So it _seems_ unnecessary to call load results success here - it muddies the logic imo and what should happen when and experiment is successfully updated.

i.e just have:


```
updateExperimentSuccess: async ({ experiment }) => {
            actions.updateExperiments(experiment)
            if (experiment.start_date) {
                actions.loadMetricResults()
                actions.loadSecondaryMetricResults()
            }
            // nothing else
})
```

Reply:
@neilkakkar Forgive me if I'm missing something obvious, but how is the state cleared from `metricResults` and `secondaryMetricResults`?

Reply:
[not saying what you're doing here wouldn't work, just trying to understand if there's a need for it. If there is, it _seems_ like an underlying bug we should fix, like some state action not cleaning up after itself.]

Reply:
> just trying to understand if there's a need for it.

Yeah. I don't recall why I added it originally but it could've been https://github.com/PostHog/posthog/pull/27391/commits/56e33b5ed0a0ac7baf548bb770e282304b745665 and I just assumed it was better to clear the results state when updating things.

https://github.com/PostHog/posthog/pull/27391 seems to work fine